academy/dao/impl/jdbc/CartDaoJDBCImpl.java

```diff
@@ -0,0 +1,125 @@
+package mate.academy.dao.impl.jdbc;
+
+import mate.academy.dao.CartDao;
+import mate.academy.model.Cart;
+import mate.academy.model.Product;
+import mate.academy.model.User;
+import mate.academy.utils.DAOService;
+import org.apache.log4j.Logger;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class CartDaoJDBCImpl implements CartDao {
+
+    private static final Logger logger = Logger.getLogger(UserDaoJDBCImpl.class);
+    private static final DAOService daoService = new DAOService();
+    private static final String SQL_ADD_PRODUCT_TO_CART = "INSERT INTO cart_product(product_id, " +
+            "cart_id) values (?, ?)";
+    private static final String SQL_CREATE_CART = "INSERT into cart(user_id) VALUES (?)";
+    private static final String SQL_GET_ALL_PRODUCTS = "SELECT p.id, p.name," +
+            " p.description, p.price " +
+            " FROM products p" +
+            " INNER JOIN cart_product cp on p.id = cp.product_id\n " +
+            " INNER JOIN cart c on cp.cart_id = c.id\n" +
+            " WHERE c.user_id;";
+    private static final String SQL_SIZE_CART = "SELECT COUNT(*) FROM cart_product WHERE " +
+            "cart_id = (?)";
+    private static final String SQL_GET_CART_ID_BY_USER = "SELECT id from cart WHERE user_id =?";
+
+
+    @Override
+    public void createCart(User user) {
+        try (Connection connection = daoService.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(SQL_CREATE_CART);
+            preparedStatement.setLong(1, user.getId());
+            preparedStatement.executeUpdate();
+            logger.info("The shopping cart was created for user with " + user.getId() + "id");
+        } catch (SQLException e) {
+            logger.error("There was an error while creating a shopping cart");

```

Comment:
we need to write in all methods:
```suggestion
            logger.error("There was an error while creating a shopping cart", e);
```

Reply:
we need to add stack trace of exception in all catch blocks